lib/chopperbot/split.ex

```diff
@@ -112,26 +112,51 @@ defmodule Chopperbot.Split do
   ## Examples
       iex> apply_options([{"a", 300}, {"b", 400}], ["+s"])
       [{"a", 330.0}, {"b", 440.0}]
+      iex> apply_options([{"a", 100}, {"b", 200}], ["-20.5%"])
+      [{"a", 79.5}, {"b", 159.0}]
   """
   @spec apply_options(orders(), options()) :: orders()
-  def apply_options(orders, ["+service" | rest]), do: apply_options(orders, ["+s" | rest])
-  def apply_options(orders, ["+vat" | rest]), do: apply_options(orders, ["+v" | rest])
+  def apply_options(orders, options) do
+    Enum.reduce(options, orders, fn option, modified_orders ->
+      apply_option(modified_orders, option)
+    end)
+  end
+
+  defp apply_option(orders, option) do
+    Enum.map(orders, fn {name, amount} ->
+      new_amount =
+        option
+        |> get_multiplier_from_option()
+        |> Kernel.*(amount)
+        |> rounding_floating_problem()
+
+      {name, new_amount}
+    end)
+  end
 
-  def apply_options(orders, ["+s" | rest]) do
-    apply_options(
-      Enum.map(orders, fn {name, amount} -> {name, rounding_floating_problem(amount * 1.10)} end),
-      rest
-    )
+  defp get_multiplier_from_option(option) when option in ["+service", "+s"] do
+    get_multiplier_from_option("+10%")
   end
 
-  def apply_options(orders, ["+v" | rest]) do
-    apply_options(
-      Enum.map(orders, fn {name, amount} -> {name, rounding_floating_problem(amount * 1.07)} end),
-      rest
-    )
+  defp get_multiplier_from_option(option) when option in ["+vat", "+v"] do
+    get_multiplier_from_option("+7%")
   end
 
-  def apply_options(orders, []), do: orders
+  defp get_multiplier_from_option(option) do
+    regex = ~r/^(\+|-)(\d+|\d+[.]\d+)(%)$/
+
+    case Regex.run(regex, option) do
+      [^option, operator, number, "%"] ->
+        {float_number, ""} = Float.parse(number)
+
+        Kernel
+        |> apply(String.to_existing_atom(operator), [100, float_number])
+        |> Kernel./(100)
+
+      _not_matched ->
+        1
+    end
+  end
```

Comment:
What should we do with unsupported options krub?

According to the previous version, we didn't handle with them so error exception would be raised when invalid option is given. 🤔

Reply:
I think it might not be good to let the invalid option becomes `1` multiplier. 
Ideally, it should halt and return the message when the option is invalid. 
What do you think?

I created the issue here: https://github.com/flipay/chopperbot/issues/7

Reply:
Agree that we should return an error message when the option is invalid 👍 

I'll revert the behavior back to the previous version for now 🙇